### PR TITLE
Manually Remove Kubernetes 1.24 jobs

### PR DESCRIPTION
**Pull request by jeremyrickard**
This PR is intended to drop the 1.24 jobs by replicating the work done to remove the 1.23 jobs in https://github.com/kubernetes/test-infra/pull/29387. 

Notably:

* This continues the pattern of  commenting  out k8s-beta in `releng/test_config.yaml`.
* k8s beta dashboard content is commented out with TODO comments for 1.29
* Removes 1.24 jobs sig-windows

Note: there were some TODO commented out things left over from the previous attempt this that should have been undone when we created the 1.28 branch and jobs. We should double check everything when it is time to do the 1.29 jobs. 

/assign @xmudrii @jimangel @marosset 
cc https://github.com/orgs/kubernetes/teams/release-engineering

/hold for discussion if this approach is still valid

There are some follow up issues from the previous drop of 1.23 jobs that we should also follow up on:

https://github.com/kubernetes/test-infra/issues/29388
https://github.com/kubernetes/test-infra/issues/29390
https://github.com/kubernetes/sig-release/issues/2227

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml [hunk 1]
```diff
@@ -326,7 +326,7 @@ periodics:
   - 'perfDashJobType: performance'
   - 'perfDashBuildsCount: 500'
 - annotations:
-    fork-per-release-periodic-interval: 24h
+    fork-per-release-periodic-interval: ""
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:329]: This is wrong, there must be one interval left here or we're going to run into issues the next time we rotate jobs.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml [hunk 2]
```diff
@@ -393,7 +393,7 @@ periodics:
           cpu: "4"
           memory: 36Gi
 - annotations:
-    fork-per-release-periodic-interval: 24h
+    fork-per-release-periodic-interval: ""
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:396]: Same here.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml [hunk 3]
```diff
@@ -437,7 +437,7 @@ periodics:
       securityContext:
         privileged: true
 - annotations:
-    fork-per-release-periodic-interval: 24h
+    fork-per-release-periodic-interval: ""
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:440]: Same here.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml [hunk 4]
```diff
@@ -482,7 +482,7 @@ periodics:
...
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:485]: Same here.

**Review by xmudrii — CHANGES_REQUESTED** [review]
We have some issue with periodic annotations that we should sort out before merging this PR. Let's sync about that and see how should we handle that.

**Review by marosset — APPROVED** [review]
windows changes LGTM

**Review by xmcqueen — COMMENTED** [review]
/lgtm

**Pull request merged**